.github/workflows/main.yml

```diff
@@ -58,4 +58,79 @@ jobs:
         run: |
           ./gradlew --stop
       - name: 'Upload to Codecov.io'
-        uses: codecov/codecov-action@v1
\ No newline at end of file
+        uses: codecov/codecov-action@v1
+
+  publish-spock:
+    runs-on: ${{ matrix.os }}
+    needs: ['build-and-verify']
+    if: github.event_name == 'push' && github.repository == 'spockframework/spock' && github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        os: ['ubuntu-latest']
+        variant: ['2.5', '3.0']   # publish needs to be done for both versions
+        java: ['8']               # publish needs the min supported java version
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache Gradle wrapper and dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/
+            ~/.gradle/wrapper/dists
+          key: test-${{ runner.os }}-variant-${{ matrix.variant }}-gradle-${{ hashFiles('**/*.gradle', '**/*.gradle.kts', '**/gradle.properties', 'gradle/**', 'buildSrc/src/main/**') }}
+          restore-keys: |
+            test-${{ runner.os }}-variant-${{ matrix.variant }}-
+      - name: 'Set up JDK'
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: 'Gradle Version'
+        run: |
+          ./gradlew --version
+      - name: 'Publish Spock'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONATYPE_OSS_USER: ${{ secrets.SONATYPE_OSS_USER }}
+          SONATYPE_OSS_PASSWORD: ${{ secrets.SONATYPE_OSS_PASSWORD }}
+          SIGNING_PASSWORD: ${{ secrets.SIGNING_GPG_PASSWORD }}
+        run: |
+          ./gradlew --no-parallel --scan --stacktrace ghActionsPublish "-Dvariant=${{ matrix.variant }}"
+      - name: 'Stop Daemon'
+        run: |
+          ./gradlew --stop
+
+  publish-docs:
+    runs-on: ${{ matrix.os }}
+    needs: ['build-and-verify']
+    if: github.event_name == 'push' && github.repository == 'spockframework/spock' && github.ref == 'refs/heads/master'
```

Comment:
I assume that docs for real the release versions are released from the particular commits placed and master (and doesn't need to be repeated from a tag), right?